workspace.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+#
+#copyright agloo in 2016 under the WTFPL 
+#constructs and groups windows into workspaces to switch between using your favorite keybinder.
+
+NUM_WS=9
+
+help() {
+	cat << EOF
+usage: $(basename $0) [-hinp] [-g ws_num] [-m ws_num] 
+	-h: Displays this message
+	-i: Initializes workspaces and puts you in workspace 0. Should be called once in a startup script.
+	-n: Move up one workspace
+	-p: Move down one workspace
+	-g, --ws_num: go to the workspace specified by ws_num
+	-m, --ws_num: move the currently focused window to the worskpace specified by ws_num
+EOF
+	exit 1
+}
+#initializes us in workspace 0
+ws_init() {
+	mkdir /tmp/workspaces/
+	for i in {0..$NUM_WS}
+	do
+		touch /tmp/workspaces/ws"$i"
+	done
+	printf 0 > /tmp/workspaces/curr
+}
+#saves all of the workspaces 
+save_ws() {
+	curr=$(cat /tmp/workspaces/curr);
+	lsw > /tmp/workspaces/ws"$curr";
+}
+move_to_ws() {
+	ws_num=$1;
+	if [ ws_num -gt $NUM_WS ]; then
+		echo "We don't have that workspace"
+		return
+	fi
+	save_ws;
+	for wid in $(lsw); do
+		mapw -u $wid
+	done
```

Comment:
Quick hint: `mapw` can take mutliple arguments, so you could replace this whole loop with: `mapw -u $(lsw)`